utils/tensorboard_parser.py

```diff
@@ -0,0 +1,143 @@
+# Standard Library
+import argparse
+import json
+
+
+class threadId:
+    def __init__(self, tid, thread_name):
+        self.tid = tid
+        self.thread_name = thread_name
+
+
+class processId:
+    def __init__(self, id, name):
+        self.id = id
+        self.name = name
+        self._threads = dict()
+
+    def add_thread(self, threadid, thread_name):
+        self._threads[threadid] = threadId(threadid, thread_name)
+
+    def get_thread_info(self, threadid):
+        return self._threads[threadid]
+
+
+class XEvent:
+    def __init__(self, ts, name, dur, pid, tid, event_args):
+        self.start_time = ts
+        self.event_name = name
+        self.duration = dur
+        self.end_time = self.start_time + self.duration
+        self.pid = pid
+        self.tid = tid
+        self.event_args = event_args
+
+
+class tensorboardEvents:
+    def __init__(self, trace_file):
+        self.trace_json_file = trace_file
+        self.timescale = "ns"
+        self._processes = dict()
+        self._x_events = list()
+        self._start_time_stamp = 0
+        self._start_time_known = False
+
+        try:
+            with open(trace_json_file) as json_data:
+                trace_json_data = json.load(json_data)
+        except Exception as e:
+            print(f"Error No metric file for {trace_json_file}: Exception {str(e)} ")
+            status = False
+            return
+
+        trace_events = trace_json_data["traceEvents"]
+        self.timescale = trace_json_data["displayTimeUnit"]
+
+        for event in trace_events:
+            if "ph" not in event:
+                continue
+            phase_type = event["ph"]
+            if phase_type == "M":
+                id = event["pid"]
+                if event["name"] == "process_name":
+                    name = event["args"]["name"]
+                    self._processes[id] = processId(id, name)
+                if event["name"] == "thread_name":
+                    name = event["args"]["name"]
+                    t_id = event["tid"]
+                    self._processes[id].add_thread(t_id, name)
+                event_args = event["args"] if "args" in event else None
+                if self._start_time_known is False:
+                    if event_args and "value" in event_args:
```

Comment:
look for specific key in args. 
{Keyname:"abc", "value":1} , I guess args will be like that so there will be value irrespective of whether it is start_time or not.